src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -202,24 +251,52 @@ def run(
                 )
                 raise unexpected_error from e
 
+    # TODO: Populate aggregation inputs
+    def _construct_line_result(self, flow_run_info, node_run_infos, output, aggregation_inputs) -> LineResult:
+        """Construct LineResult from flow_run_info, node_run_infos and output"""
+        return LineResult(
+            output=output,
+            aggregation_inputs=aggregation_inputs,
+            run_info=flow_run_info,
+            node_run_infos=node_run_infos,
+        )
+
+    def _load_outputs(self, output_dir: Path) -> List[Dict[str, Any]]:
+        """Load the outputs of a batch run from output dir
+        :param output_dir: Output dir of a batch run
+        :type output_dir: Path
+        :return: List of output dicts
+        :rtype: List[Dict[str, Any]]
+        """
+        path = output_dir / "output.jsonl"
+        outputs = []
+        with open(path, "r", encoding="utf-8") as fin:
+            for line in fin:
+                outputs.append(json.loads(line))
+        return outputs
+
     def cancel(self):
         """Cancel the batch run"""
         self._is_canceled = True
 
     async def _exec_in_task(
         self,
-        batch_inputs: List[Dict[str, Any]],
+        full_inputs: List[Dict[str, Any]],
         run_id: str = None,
         output_dir: Path = None,
         raise_on_line_failure: bool = False,
+        previous_line_results: List[LineResult] = None,
+        lines_to_run: List[int] = None,
```

Comment:
Please remove the input parameter `lines_to_run`
It can be calculated from `full_inputs` and `previous_line_results`